power_grid_model_c/power_grid_model_cpp/CMakeLists.txt

```diff
@@ -30,6 +30,7 @@ target_sources(power_grid_model_cpp PUBLIC
 set_target_properties(power_grid_model_cpp PROPERTIES
   VERSION ${PGM_VERSION}
   SOVERSION ${PGM_VERSION}
+  CXX_STANDARD 17
```

Comment:
Will this work as intended? The idea is that the C++ API self is C++17 compliant. But it should not require the user to use C++17 only. User (like validation test) should be able to use any C++ version which is equal or higher than C++17.

Reply:
it will work cause it's the target property, not the overload `set(CMAKE_CXX_STANDARD)`. See also how the other libraries still compile, even if they use C++20 features. It does not enforce upstream/downstream dependencies.

In fact, setting the `CXX_STANDARD` property on the library still doesn't enforce the compiler to only use that interface library with C++17 features only. It's more for exposure than for actually doing something

Reply:
Okay good to know.

Reply:
ok interestingly, for MSVC, it does make a difference, but not for clang-cl. welp, clang-tidy-19 seems to use the `CXX_STANDARD` target property as well so it's all good now